modules/renter/contractor/formcontract.go

```diff
@@ -382,7 +128,7 @@ func (c *Contractor) formContracts(a modules.Allowance) error {
 		c.log.Printf("WARN: failed to form desired number of contracts (wanted %v, got %v): %v", a.Hosts, numContracts, strings.Join(errs, "\n"))
 	}
 	c.mu.Lock()
-	c.renewHeight = endHeight
+	c.renewHeight = c.blockHeight + a.Period // TODO: this risks desync
```

Comment:
`desync` here means the contracts could end at different heights than `c.renewHeight`. I'm not sure why I changed this. The old method of passing `endHeight` to `newContract` seems fine.